pkg/controller/extendeddaemonsetreplicaset/strategy/utils.go

```diff
@@ -23,9 +23,29 @@ import (
 	datadoghqv1alpha1 "github.com/datadog/extendeddaemonset/pkg/apis/datadoghq/v1alpha1"
 	"github.com/datadog/extendeddaemonset/pkg/controller/extendeddaemonsetreplicaset/conditions"
 	podaffinity "github.com/datadog/extendeddaemonset/pkg/controller/utils/affinity"
+	"github.com/datadog/extendeddaemonset/pkg/controller/utils/comparison"
 	podutils "github.com/datadog/extendeddaemonset/pkg/controller/utils/pod"
 )
 
+func compareCurrentPodWithNewPod(edsName string, replicaset *datadoghqv1alpha1.ExtendedDaemonSetReplicaSet, pod *corev1.Pod, node *corev1.Node) bool {
+	// check that the pod correspond to the replicaset. if not return false
+	if !compareSpecTemplateMD5Hash(replicaset.Spec.TemplateGeneration, pod) {
+		return false
+	}
+	if !compareNodeResourcesAnnotationsMD5Hash(edsName, replicaset, pod, node) {
+		return false
+	}
+	return true
+}
+
+func compareNodeResourcesAnnotationsMD5Hash(edsName string, replicaset *datadoghqv1alpha1.ExtendedDaemonSetReplicaSet, pod *corev1.Pod, node *corev1.Node) bool {
```

Comment:
same question, is there a risk of receiving nil pointers here ?

Reply:
I don't think so